sdk/storage/Azure.Storage.Common/src/Shared/StorageSharedKeyPipelinePolicy.cs

```diff
@@ -162,5 +161,102 @@ private void BuildCanonicalizedResource(StringBuilder stringBuilder, Uri resourc
+                0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,
+                0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,
+                0x0,  0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12,
+                0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x12, 0x0,  0x0,  0x0,  0x0,  0x0,
+                0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,
+                0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,  0x0,
+            };
+
+            private static readonly int[] s_table_lv4 =
+            {
+                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0,
+                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0,
+                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x8012, 0x0, 0x0, 0x0, 0x0, 0x0, 0x8212, 0x0, 0x0,
+                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0,
+                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0,
+                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0,
+                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0,
+                0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0, 0x0, 0x0, 0x0, 0x0,    0x0, 0x0,
+            };
+
+            private static readonly int[][] s_tables = { s_table_lv0, s_table_lv2, s_table_lv4 };
+
+            public int Compare(string x, string y)
+            {
+                int currentLevel = 0;
+                int i = 0;
+                int j = 0;
+
+                while (currentLevel < s_tables.Length)
+                {
+                    if (currentLevel < s_tables.Length - 1 && i != j)
+                    {
+                        return i - j;
+                    }
+
+                    int weight1 = i < x.Length ? s_tables[currentLevel][x[i]] : 0x1;
+                    int weight2 = j < y.Length ? s_tables[currentLevel][y[j]] : 0x1;
+
+                    if (weight1 == 0x1 && weight2 == 0x1)
+                    {
+                        i = 0;
+                        j = 0;
+                        currentLevel++;
+                    }
+                    else if (weight1 == weight2)
+                    {
+                        i++;
+                        j++;
+                    }
+                    else if (weight1 == 0)
+                    {
+                        i++;
+                    }
+                    else if (weight2 == 0)
+                    {
+                        j++;
+                    }
+                    else
+                    {
+                         return weight1 - weight2;
```

Comment:
I think this line may be the issue? In C++ its `return weight1 < weight2` but since we need to switch to int here, this operation is backwards, right? I'm mostly comparing to line 277 where you have `i > j` being translated to `i - j` but since the is less than, one of these is backwards I think.

Reply:
It may actually be line 277 that's backwards when compared to Python.
https://github.com/Azure/azure-sdk-for-python/blob/1677cdbcb9ae34bc02a0278d79aadc5af9c1341b/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py#L62-L66

Either way, I think one subtraction op is backwards.

Reply:
Okay after futher testing, I made this change in Python, [PR here](https://github.com/Azure/azure-sdk-for-python/pull/36552):
![image](https://github.com/user-attachments/assets/7f0bf427-8844-4c20-b1a7-2a97b2f39740)

And my testcases are still passing. So I do believe that the issue lies in lines 225-228 in this PR. I have the suggested change linked in the above PR, but will leave a comment above

Reply:
@jalauzon-msft, all tests fail if I change like 254 to `return weight2 - weight 1`